packages/@tailwindcss-upgrade/src/migrate-js-config.ts

```diff
@@ -196,15 +196,38 @@ async function migrateTheme(
 }
 
 function migrateDarkMode(unresolvedConfig: Config & { darkMode: any }): string {
-  let variant: string = ''
+  let variant: string | string[] = ''
   let addVariant = (_name: string, _variant: string) => (variant = _variant)
   let config = () => unresolvedConfig.darkMode
   darkModePlugin({ config, addVariant })
 
   if (variant === '') {
     return ''
   }
-  return `\n@tw-bucket custom-variant {\n@custom-variant dark (${variant});\n}\n`
+
+  if (!Array.isArray(variant)) {
+    variant = [variant]
+  }
+
+  if (variant.length === 1 && !variant[0].includes('{')) {
+    return `\n@tw-bucket custom-variant {\n@custom-variant dark (${variant[0]});\n}\n`
+  }
+
+  let customVariant = ''
+  for (let variantName of variant) {
+    // Convert to the block syntax if a block is used
+    if (variantName.includes('{')) {
+      customVariant += variantName.replace('}', '{ @slot }}') + '\n'
+    } else {
+      customVariant += variantName + '{ @slot }\n'
+    }
```

Comment:
Not 100% if this is enough, can we add a few more test cases:

Nested media queries:
```
darkMode: ['variant', '@media not print { @media screen { .dark & } }'],
```
```css
/* dark:flex */
@media not print {
  @media screen {
    .dark .dark\:flex {
      display: flex;
    }
  }
}
```
Play: https://play.tailwindcss.com/n3k65J6HJF?file=config